repos/system_upgrade/common/libraries/rpms.py

```diff
@@ -114,3 +115,36 @@ def check_file_modification(config):
     """
     output = _read_rpm_modifications(config)
     return _parse_config_modification(output, config)
+
+
+def get_leapp_packages(source_version=None, component=None, include_deps=False):
+    """
+    Get list of leapp packages.
+
+    :param source_version: a list or string specifying source_versions. If not defined then all packages
+        for current source_target_version will be returned.
+    :param component: a list or string specifying leapp components. If defined then only packages related
+        to the specific component will be returned.
+    :param include_deps: a flag to control whether deps meta packages are to be filtered out.
+    """
+    res = set()
+    # NOTE(ivasilev) mapping has the format component: {source_version: [pkg1, pkg2, ..]}
+    component_map = {'framework': {'7': ['leapp', 'leapp-deps', 'python2-leapp'],
+                                   '8': ['leapp', 'leapp-deps', 'python3-leapp']},
+                     'repository': {'7': ['leapp-upgrade-el7toel8', 'leapp-upgrade-el7toel8-deps'],
+                                    '8': ['leapp-upgrade-el8toel9', 'leapp-upgrade-el8toel9-deps']},
+                     'snactor': {'7': ['snactor'],
+                                 '8': ['snactor']}}
+    source_versions = [source_version] if isinstance(source_version, str) else (source_version or
+                                                                                [get_source_major_version()])
+    components = [component] if isinstance(component, str) else component or list(component_map.keys())
+    for comp in components:
+        for src_version in source_versions:
+            try:
+                res.update(component_map[comp][src_version])
+            except KeyError:
+                continue
```

Comment:
Do we really want to raise an exception in case user requests wrong version or component? Imho an empty list of packages would make it pretty obvious that something is wrong.

Reply:
Alright, let's make it an exception then.

Reply:
My criteria for an exception vs something that would pass silently would be something like:

* If there is an error then an exception should be raised.

* If there is a cornercase that is not an error, then an "empty" value like empty list can (and likely should) be returned.

  An example that shows both of the above criteria in action would be:
  ``` python
    def make_hrefs((list_of_urls):
        hrefs = []
        for url in list_of_urls:
            if not validate_url(url):
                raise ValueError(f"{url} was not a valid url")
            hrefs.append(f"<a href={url}>{url}</a>")
        return hrefs
  ```
In the code in this PR, we have a small set of valid user inputs so I think it should be an error if the programmer calling this function sends an invalid input.  That way they are quickly notified that they have a bug that they need to fix.  If it's not an error then a programmer could make a typo when calling the function and either not notice there was a problem or it could cause an error far away from where the typo was made (leading to a more difficult time debugging the problem).

* Once in a while I see or write code that returns a value that isn't a possible, valid value (usually `None`) when it encounters a cornercase.  This probably happens for things that are deemed not to be errors when the return value is normally a scalar (example: an int) Since any integer returned might be valid, another value needs to be chosen.  I think these should be evaluated very closely since they can lead to errors in caller's code.  For instance, imagine if `strcmp()` was written to return `None` if the input value was not a string.  Then this code would be buggy but might not appear so in testing:
  ``` python
    value1 = str(100)
    value2 = 99  # Programmer forgot to convert this value
    # Programmer knows strcmp returns 0 when strings are equal and 1 or -1 if they aren't
    # but they forget that it returns None when bad values are given.
    if not strcmp(value1, value2):
        do_thing_when_strings_are_equal()
  ```
  I think returning a sentinel value is common in code influenced by C since C doesn't have exceptions but it should be avoided when possible in Python code.